src/VisualStudio/Core/Impl/CodeModel/CodeModelIncrementalAnalyzer.cs

```diff
@@ -89,7 +89,7 @@ public void FireEvents(DocumentId documentId, CancellationToken cancellationToke
                         return false;
                     }
 
-                    if (!codeModelProvider.ProjectCodeModel.TryGetCachedFileCodeModel(filename, out var fileCodeModelHandle))
+                    if (!((ProjectCodeModel)codeModelProvider.ProjectCodeModel).TryGetCachedFileCodeModel(filename, out var fileCodeModelHandle))
```

Comment:
i don't understand this cast.

Reply:
See long comment on IProjectCodeModelProvider.cs. (But good question.)